src/modules/shared/pages/innovation/sections/section-info.component.ts

```diff
@@ -52,169 +40,188 @@ export class PageInnovationSectionInfoComponent extends CoreComponent implements
   shouldShowDocuments = false;
 
+  sectionInfo: {
+    id: string,
+    nextSectionId: null | string,
+    title: string,
+    status: { id: keyof typeof INNOVATION_SECTION_STATUS, label: string },
+    submitButton: { show: boolean, label: string },
+    isNotStarted: boolean,
+    hasEvidences: boolean,
+    wizard: WizardEngineModel,
+    allStepsList: InnovationSectionStepLabels,
+    date: string,
+    submittedBy: null | { name: string, isOwner?: boolean },
+    openTasksCount: number
+  };
+  
+
   constructor(
     private activatedRoute: ActivatedRoute,
     private innovationDocumentsService: InnovationDocumentsService
   ) {
 
     super();
 
+    this.sectionId = this.activatedRoute.snapshot.params.sectionId;
     this.innovation = this.stores.context.getInnovation();
 
-    this.section = {
-      id: this.activatedRoute.snapshot.params.sectionId,
+    this.sectionsIdsList = getInnovationRecordConfig().flatMap(sectionsGroup => sectionsGroup.sections.map(section => section.id));
+
+    this.baseUrl = `${this.stores.authentication.userUrlBasePath()}/innovations/${this.innovation.id}`;
+    
+    // Flags
+    this.isInnovatorType = this.stores.authentication.isInnovatorType();
+    this.isAccessorType = this.stores.authentication.isAccessorType();
+    this.isAssessmentType = this.stores.authentication.isAssessmentType();
+
+    this.sectionInfo = {
+      id: '',
       nextSectionId: null,
       title: '',
       status: { id: 'UNKNOWN', label: '' },
-      isNotStarted: false,
       submitButton: { show: false, label: "Confirm section answers" },
+      isNotStarted: false,
       hasEvidences: false,
       wizard: new WizardEngineModel({}),
+      allStepsList: {},
       date: '',
       submittedBy: null,
       openTasksCount: 0
     };
 
-    this.sectionsIdsList = getInnovationRecordConfig().flatMap(sectionsGroup => sectionsGroup.sections.map(section => section.id));
-
-    this.baseUrl = `${this.stores.authentication.userUrlBasePath()}/innovations/${this.innovation.id}`;
-
-    // Flags
-    this.isInnovatorType = this.stores.authentication.isInnovatorType();
-    this.isAccessorType = this.stores.authentication.isAccessorType();
-    this.isAssessmentType = this.stores.authentication.isAssessmentType();
-
   }
 
   ngOnInit(): void {
-
+    
     this.initializePage();
 
     // This router subscription is needed for the button to go to the next step.
     // As is it the same component, we can't use the routerLink directive alone.
+    
     this.subscriptions.push(
-      this.router.events.pipe(filter((e): e is NavigationEnd => e instanceof NavigationEnd)).subscribe(e => this.initializePage())
-    );
-
-  }
-
 
-  private getNextSectionId(): string | null {
-
-    const currentSectionIndex = this.sectionsIdsList.indexOf(this.section.id);
-
-    return this.sectionsIdsList[currentSectionIndex + 1] || null;
+      this.activatedRoute.params.subscribe(() => {
+        this.initializePage();
```

Comment:
Here I think you don't need to subscribe to activatedRoute params as you have on line 109 a subscription that takes care of it.

Reply:
You are right, it does! But it wasnt, for some weird reason, updating things properly before, that's why I added that subscription. 
Funny enough, now I removed it and is working as intended.  x_x